sdk/core/azure-core/azure/core/tracing/ext/opencensus.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+from opencensus.trace import execution_context
+from opencensus.trace import tracer as tracer_module, Span
+from opencensus.trace.propagation import trace_context_http_header_format
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracers.noop_tracer import NoopTracer
+
+
+class OpencensusWrapper:
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Any) -> None
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None:
+            instrumentation_key = self._get_environ("APPINSIGHTS_INSTRUMENTATIONKEY")
+            prob = self._get_environ("AZURE_TRACING_SAMPLER") or 0.001
+            if tracer is None or isinstance(tracer, NoopTracer):
+                if instrumentation_key is not None:
+                    from opencensus.ext.azure.trace_exporter import AzureExporter
+
+                    tracer = tracer_module.Tracer(
+                        exporter=AzureExporter(instrumentation_key=instrumentation_key),
+                        sampler=ProbabilitySampler(prob),
+                    )
+                else:
+                    tracer = tracer_module.Tracer(sampler=ProbabilitySampler(prob))
+                self.was_created_by_azure_sdk = True
+            span = tracer.span(name=name)
+
+        self.tracer = tracer
+        self.trace_id = None
+        if span.context_tracer:
+            self.trace_id = span.context_tracer.span_context.trace_id
+        self.span_instance = span
+        self.span_id = str(span.span_id)
```

Comment:
same comment about a property method to delagate to the span instance rather than copying from it

Reply:
I just got rid of trace and span id as I never really used it.